code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -366,13 +366,12 @@
 	var/mob/living/simple_animal/hostile/H = from
 	if(istype(H))
 		for (var/datum/weakref/ref in H.friends)
-			var/not_a_friend_yet = TRUE
+			var/already_friend = FALSE
 			var/mob/M = ref.get()
-			for (var/datum/weakref/reff in H.friends)
+			for (var/datum/weakref/reff in friends)
```

Comment:
This is the only change that does anything. The rest have no effect, except the removal of the `break`, which reduces performance.